operator/pkg/controllers/master/master.go

```diff
@@ -69,28 +77,76 @@ func (c *Controller) createSecrets(ctx context.Context, controlPlane *v1alpha1.C
 	return nil
 }
 
-func certListFor(controlPlane *v1alpha1.ControlPlane) []*secrets.Request {
+func (c *Controller) getControlPlaneHostname(ctx context.Context, controlPlane *v1alpha1.ControlPlane) (string, error) {
+	hostname, err := c.createService(ctx, controlPlane)
+	if err != nil {
+		return "", err
+	}
+	if hostname == "" {
+		return "", fmt.Errorf("waiting for control plane hostname, %w", errors.WaitingForSubResources)
```

Comment:
In the long run, let's model this as a reconcile.result instead of munging it into a typed error.